packages/lodestar/src/network/peers/discover.ts

```diff
@@ -357,7 +357,7 @@ export class PeerDiscovery {
    * Peers that have been returned by discovery requests are dialed here if they are suitable.
    */
   private async dialPeer(cachedPeer: CachedENR): Promise<void> {
-    this.peersToConnect--;
+    this.peersToConnect = Math.max(this.peersToConnect - 1, 0);
```

Comment:
more of a question for my own understanding. In the attached screenshot, the value of peersToConnect  was in the range of -6000. Question is, should that be happening in the first place? Because this fix might be fixing the symptoms while the core issue of continually calling `dialPeer` which decrease `peersToConnect` till it enters negative value is not fixed...

Reply:
we dial a peer when
- `this.peersToConnect > 0`
- or the peer subscribes to a subnet that we want

if `this.peersToConnect` is 3 while we need to dial 5 subnet peers, in that case we want `this.peersToConnect` to be 0 instead of a negative value. The next heartbeat may increase `this.peersToConnect` again if some dials are not successful